### PR TITLE
IBM OpenShift example Action

### PR DESCRIPTION
**Pull request by jjasghar**
- A basic example for IBM Cloud OpenShift Deployment
- Installs the IBM Cloud CLI
- Installs a version of OC to interface with OpenShift
- Runs a create and deploy in OpenShift

Signed-off-by: JJ Asghar <jjasghar@gmail.com>
Signed-off-by: JJ Asghar <awesome@ibm.com>

In the workflow and properties files:

- [x] The workflow filename of CI workflows should be the name of the language or platform, in lower case.  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").

  The workflow filename of publishing workflows should be the name of the language or platform, in lower case, followed by "-publish".
- [x] Includes a matching `ci/properties/*.properties.json` file.
- [x] Use sentence case for the names of workflows and steps, for example "Run tests".
- [x] The name of CI workflows should only be the name of the language or platform: for example "Go" (not "Go CI" or "Go Build")
- [x] Include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] CI workflows should run on `push` to `branches: [ master ]` and `pull_request` to `branches: [ master ]`.

  Packaging workflows should run on `release` with `types: [ created ]`.

Some general notes:

- [x] This workflow must only use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**

  This workflow must only use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  Workflows using these actions must reference the action using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] This workflow must not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] This workflow must not use a paid service or product.

### CODE REVIEW

**Comment by SolutionWala** [comment]
Shift +I

**Comment by vkkghjkgg** [comment]
Losiver

**Comment by jjasghar** [comment]
Not sure why the CI error failed: https://github.com/actions/starter-workflows/pull/489/files#diff-4311fc13e12ea7f16e69350c90eb9ddd

https://github.com/actions/starter-workflows/pull/489/checks?check_run_id=713891210#step:4:7

The file is there.

**Comment by vkkghjkgg** [comment]
انا لم أقف شيئاً عليك انتا بربط الأمر
عليك انتا التاسيس الرب

في الأربعاء، ٢٧ مايو، ٢٠٢٠ ١٨:٥٤ JJ Asghar <notifications@github.com> كتب:

File: ci/ibm-openshift.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# This workflow will build a docker container, publish and deploy it to IBM's version of OpenShift
+#
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:2]: Losiver

File: ci/ibm-openshift.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+# To configure this workflow:
+#
+# 1. Ensure that your repository contains a Dockerfile
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:5]: Losiver

File: ci/ibm-openshift.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+# 2. Setup secrets in your repository by going to settings: Create IBM_CLOUD_API_KEY
+# 3. Change the values for the IMAGE_NAME, OS_CLUSTER, PROJECT, and OC_VERSION and OC_VERSION_HASH
+# 3.a: Please check: https://www.okd.io/download.html to see the newest version of the oc client
+
+name: Build and Deploy to IBM OpenShift
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:10]: Losiver

File: ci/ibm-openshift.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+on:
+  release:
+    types: [created]
+
+# Environment variables available to all jobs and steps in this workflow
+env:
+  GITHUB_SHA: ${{ github.sha }}
+  IBM_CLOUD_API_KEY: ${{ secrets.IBM_CLOUD_API_KEY }}
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:19]: Losiver

File: ci/ibm-openshift.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  IBM_CLOUD_REGION: us-south
+  IMAGE_NAME: node:10
+  OS_CLUSTER: example-iks-cluster-name-or-id
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:22]: Losiver

File: ci/ibm-openshift.yml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  PROJECT: hello-word
+  OC_VERSION: 3.11.0
+  OC_VERSION_HASH: 0cbc58b
+
+jobs:
+  setup-build-publish-deploy:
+    name: Setup, Build, Publish, and Deploy
+    runs-on: ubuntu-latest
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:30]: Losiver

File: ci/ibm-openshift.yml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v2
+
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:35]: Losiver

File: ci/ibm-openshift.yml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    # Verify/Install some dependancies
+    - name: Install or Verify some dependancies
+      run: |
+        apt-get update
+        if ! [ -x "$(command -v curl)" ]; then
+          echo 'Error: curl is not installed, or not in path.' >&2
+          echo 'Installing curl'
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:42]: Losiver

File: ci/ibm-openshift.yml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+          apt-get install -y curl
+        fi
+        if ! [ -x "$(command -v jq)" ]; then
+          echo 'Error: jq is not installed, or not in path.' >&2
+          echo 'Installing jq'
+          apt-get install -y jq
+        fi
+
```
Comment (vkkghjkgg) [inline ci/ibm-openshift.yml:50]: Losiver

File: ci/properties/ibm-openshift.json [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "Build and Deploy to IBM OpenShift",
+    "description": "Build a docker container, publish it, and deploy it the IBM OpenShift offering.",
```
Comment (vkkghjkgg) [inline ci/properties/ibm-openshift.json:3]: Losiver

**Comment by cschleiden** [comment]
> Not sure why the CI error failed: https://github.com/actions/starter-workflows/pull/489/files#diff-4311fc13e12ea7f16e69350c90eb9ddd
> 
> https://github.com/actions/starter-workflows/pull/489/checks?check_run_id=713891210#step:4:7
> 
> The file is there.

@jjasghar 👋 
The file in this PR is called `ibm-openshift.json` but we expect it to be named `ibm-openshift.properties.json` :)

**Comment by jjasghar** [comment]
@cschleiden do'h fixed. thanks :)

File: ci/ibm-openshift.yml [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+# Environment variables available to all jobs and steps in this workflow
+env:
+  GITHUB_SHA: ${{ github.sha }}
+  IBM_CLOUD_API_KEY: ${{ secrets.IBM_CLOUD_API_KEY }}
+  IBM_CLOUD_REGION: us-south
+  IMAGE_NAME: node:10
+  OS_CLUSTER: example-iks-cluster-name-or-id
+  PROJECT: hello-word
+  OC_VERSION: 3.11.0
```
Comment (stevemar) [inline ci/ibm-openshift.yml:24]: use 4.3, not 3.11

Comment (jjasghar) [reply ci/ibm-openshift.yml:24]: 🤦

**Comment by andymckay** [comment]
Noting that the open shift workflow has landed: https://github.com/github/c2c-actions/issues/2002

**Pull request closed without merging**